apps/src/gamelab/AnimationPicker/animationPickerModule.js

```diff
@@ -98,34 +99,76 @@ exports.beginUpload = function (filename) {
  * @param {!{filename: string, result: number, versionId: string}} result
  * @returns {function}
  */
-exports.handleUploadComplete = function (result) {
+export function handleUploadComplete(result) {
   return function (dispatch, getState) {
     var state = getState().animationPicker;
     var goal = state.goal;
     var uploadFilename = state.uploadFilename;
     if (goal === Goal.NEW_ANIMATION) {
-      dispatch(gamelabActions.addAnimation({
-        key: result.filename.replace(/\.png$/i, ''),
-        name: uploadFilename,
-        size: result.size,
-        version: result.versionId
-      }));
+      // TODO (bbuchanan): This sequencing feels backwards.  Eventually, we
+      // ought to preview and get dimensions from the local filesystem, async
+      // with the upload itself, but that will mean refactoring away from the
+      // jQuery uploader.
+      loadImageMetadata(result, uploadFilename, animation => {
+        dispatch(gamelabActions.addAnimation(animation));
+      });
```

Comment:
I've left a TODO here but it's not something I'd like to deal with in this PR.

Right now the flow goes something like this:

1. Select file
2. Upload file to S3
3. Download file from S3
4. Load downloaded file as Image to get its dimensions
5. Save image metadata to S3

Eventually I'd prefer it look like this:

1. Select file
2. Load file from disk as Image to get its dimensions (and validate!)
3. Upload file to S3
4. Upload image metadata to S3

That's going to be a chunk of work though; I'm not sure there's a hook in the jQuery uploader plugin we're currently using to do an async operation like loading the image from disk between picking the file and uploading the file.